packages/pages/src/generate/ci/ci.ts

```diff
@@ -0,0 +1,94 @@
+import path from "path";
+import { CommandModule } from "yargs";
+import { Path } from "../../common/src/project/path.js";
+import { defaultProjectStructureConfig } from "../../common/src/project/structure.js";
+import { CiConfig, Plugin } from "../../common/src/ci/ci.js";
+import fs from "node:fs";
+import colors from "picocolors";
+
+const handler = (): void => {
+  const ciConfigFilename =
+    defaultProjectStructureConfig.filenamesConfig.ciConfig;
+  const sitesConfigRoot =
+    defaultProjectStructureConfig.filepathsConfig.sitesConfigRoot;
+  const ciConfigAbsolutePath = new Path(
+    path.join(process.cwd(), sitesConfigRoot, ciConfigFilename)
+  );
+  updateCiConfig(ciConfigAbsolutePath.getAbsolutePath(), true);
+};
+
+export const ciCommandModule: CommandModule<unknown, unknown> = {
+  command: "ci",
+  describe: "Generates ci.json file",
+  builder: (yargs) => yargs,
+  handler,
+};
+
+export const updateCiConfig = (
+  ciConfigPath: string,
+  calledViaCommand: boolean
+) => {
+  const ciDir = path.dirname(ciConfigPath);
+  if (!fs.existsSync(ciDir)) {
+    fs.mkdirSync(ciDir);
+  }
+
+  let originalCiConfigJson = {} as CiConfig;
+  if (fs.existsSync(ciConfigPath)) {
+    originalCiConfigJson = JSON.parse(fs.readFileSync(ciConfigPath).toString());
+  } else {
+    if (calledViaCommand) {
+      console.error(colors.red("sites-config/ci.json is missing"));
+      return;
+    } else {
+      throw new Error("sites-config/ci.json is missing");
+    }
+  }
+
+  const ciJson = mergeCiConfig(originalCiConfigJson);
+  if (ciJson) {
+    fs.writeFileSync(ciConfigPath, JSON.stringify(ciJson, null, "  "));
+  }
+};
+
+export const mergeCiConfig = (ciConfig: CiConfig): CiConfig | undefined => {
```

Comment:
rather than mutate the existing config -- thoughts on just copying it and returning a new one?